### ISSUE TITLE
[QST]: Auto scale parameter for python quadtree point in polygon?

### ISSUE THREAD

**Issue by thomcom** [rapidsai/cuspatial#900]
### What is your question?

Hey @zhangjianting @trxcllnt how do you feel about automatically computing the scale parameter in the implementation of `.contains_properly`? I was thinking of making it a ratio of the mean standard deviation of the x and y points that are going to be used to create the quadtree. If I used the mean of the x and y standard deviations, would it be better to use the mean unmodified or use $\frac{mean}{10}$ instead?

**Comment by harrism** [comment]
Wouldn't computing the mean and std be a significant cost? (O(N))

**Comment by thomcom** [comment]
I don't think it would be very significant, given the number of gathers and scatters that occur in order to preprocess and postprocess the data for `contains_properly`. There's significant overhead to put the data into the right format to look like GeoPandas and have better usability. However the cost of the main operation, point in polygon, should still be insignificant compared to GeoPandas so the overall speedup will remain substantial. At the moment there's more than 350ms of overhead in aligning indexes and type checking from preprocessing a 13m point dataset (see #892), so pip takes (estimating) 40x longer than using it in its most granular form. Adding another set of O(N) gpu operations won't increase that overhead substantially imo.

**Comment by harrism** [comment]
We should be eliminating Python overhead, not adding to it.  Sounds like more work should be pushed to device code or at least C++ to make this more efficient.

**Comment by trxcllnt** [comment]
Is determining the scale this way something that could be done by the user? If so, it seems like a good candidate to put in docs, or maybe even add a function that they can call beforehand.

**Comment by thomcom** [comment]
Sorry, the title is misleaing. I'm not proposing that we add auto scaling to the fundamental quadtree point in polygon operations, just to its use with `.contains_properly`.
